modules/evm/src/lib.rs

```diff
@@ -1606,7 +1606,8 @@ impl<T: Config> Pallet<T> {
 			caller, user, limit, amount
 		);
 
-		T::Currency::reserve_named(&RESERVE_ID_STORAGE_DEPOSIT, &user, amount)
+		T::ChargeTransactionPayment::reserve_fee(&user, amount, Some(RESERVE_ID_STORAGE_DEPOSIT))?;
```

Comment:
don't we need to do the same for unreserve?

Reply:
As I comment in the beginning:
> we do not need to change [unreserve_storage](https://github.com/AcalaNetwork/Acala/blob/7c16ceca094f47d8dea49c000f1c0f169c20b641/modules/evm/src/lib.rs#L1630) and charge_storage in evm, because reserve_storage happend before those two operation. That is, unreserve_storage and [charge_storage](https://github.com/AcalaNetwork/Acala/blob/7c16ceca094f47d8dea49c000f1c0f169c20b641/modules/evm/src/lib.rs#L1656-L1668) always have native token.

In other words, If user don't have enough native token, and `T::ChargeTransactionPayment::reserve_fee` success, it makes user got enough reserved native token. then `unreserve` based on native token will always success. of course, the amount of `unreserve` should not large then `reserve`, but that's invoker's duty.

Reply:
it's going to be the same but for consistency `reserve_fee` and `un reserve_fee`